### PR TITLE
test: migrate away from jest

### LINKED ISSUE #132: Migrate to `tap@18` from `jest`
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

_No response_

### PR DESCRIPTION
**Pull request by puskin**
Closes https://github.com/fastify/fastify-awilix/issues/132

I have no idea what you guys would like to have to check the coverage, so I opted for `c8`, requiring 0 configs files and being plug-n-play

### CODE REVIEW

**Comment by gurgunday** [comment]
c8 is what we (and tap) use, thank you for this PR!

**Comment by gurgunday** [comment]
Just make sure to target `next`

https://github.com/fastify/fastify-awilix/tree/next

**Comment by puskin94** [comment]
> Just make sure to target `next`
> 
> https://github.com/fastify/fastify-awilix/tree/next

uh, nice catch! just updated. 

Targetting node 18 now, should we skip the jobs for node 14 an 16, being failing?

**Comment by gurgunday** [comment]
> Targetting node 18 now, should we skip the jobs for node 14 an 16, being failing?

Yes, it will test for node >=18 in `next`, as soon as the conflicts are resolved

**Comment by puskin94** [comment]
> > Targetting node 18 now, should we skip the jobs for node 14 an 16, being failing?
> 
> Yes, it will test for node >=18 in `next`, as soon as the conflicts are resolved

Yeah, that was my bad :)

Should be fixed now!

**Review by simoneb — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -13,9 +13,9 @@
   "type": "commonjs",
   "types": "lib/index.d.ts",
   "scripts": {
-    "test": "jest --config=jest.config.json",
+    "test": "node --test",
+    "test:coverage": "c8 --100 npm run test",
```
Comment (Fdawgs) [inline package.json:17]: Please move this line back, the `test:` scripts were sorted alphabetically.

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by marco-ippolito — APPROVED** [review]

**Review by Fdawgs — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -13,9 +13,9 @@
   "type": "commonjs",
   "types": "lib/index.d.ts",
   "scripts": {
-    "test": "jest --config=jest.config.json",
+    "test": "node --test",
```
Comment (gurgunday) [inline package.json:16]: Coverage should be enabled by default as is already the case with `tap@18` in other repos

Comment (gurgunday) [reply package.json:16]: The CI only runs `npm run test` so currently it doesn't check coverage

Comment (gurgunday) [reply package.json:16]:
```suggestion
    "test": "c8 node --test",
```

File: package.json [hunk 3]
```diff
@@ -13,9 +13,9 @@
...
     "test:ci": "npm run lint && npm run test:typescript && npm run test:coverage",
-    "test:coverage": "jest --config=jest.config.json --coverage",
+    "test:coverage": "c8 --100 npm run test",
```
Comment (gurgunday) [inline package.json:18]: I think you can delete this after https://github.com/fastify/fastify-awilix/pull/134/files#r1573219656

**Review by gurgunday — COMMENTED** [review]
Hopefully the last review ;)

You forgot the most important step, removing `jest`

https://github.com/fastify/fastify-awilix/blob/fd7bde2bd1ba2b8cde997725586af7ce2898a2b2/package.json#L40

**Comment by puskin94** [comment]
> Hopefully the last review ;)
> 
> You forgot the most important step, removing `jest`
> 
> https://github.com/fastify/fastify-awilix/blob/fd7bde2bd1ba2b8cde997725586af7ce2898a2b2/package.json#L40

oh boi, I didn't forget, it got reintroduced when I merged "next" into my branch!

Good catch, and thanks :)

**Review by gurgunday — APPROVED** [review]
lgtm

**Comment by gurgunday** [comment]
Thank you!

**Pull request merged**